engine/src/main/java/io/zeebe/engine/processor/workflow/job/JobBatchActivateProcessor.java

```diff
@@ -37,13 +37,23 @@
   private final JobState jobState;
   private final VariablesState variablesState;
   private final KeyGenerator keyGenerator;
+  private final long maxJobBatchLength;
+
   private final ObjectHashSet<DirectBuffer> variableNames = new ObjectHashSet<>();
 
   public JobBatchActivateProcessor(
-      JobState jobState, VariablesState variablesState, KeyGenerator keyGenerator) {
+      JobState jobState,
+      VariablesState variablesState,
+      KeyGenerator keyGenerator,
+      long maxRecordLength) {
+
     this.jobState = jobState;
     this.variablesState = variablesState;
     this.keyGenerator = keyGenerator;
+
+    // we can only add the half of the max record length to the job batch
+    // because the jobs itself are also written to the same batch
+    maxJobBatchLength = (maxRecordLength - Long.BYTES) / 2;
```

Comment:
Why minus long?

Reply:
I guess it is because of the fragment alignment :sweat_smile:

Reply:
:see_no_evil: